app/Http/Controllers/ServidorController.php

```diff
@@ -22,52 +22,64 @@ public function index()
     public function create()
     {
         $cargos = Cargo::all();
-        $tipo_usuarios = TipoUsuario::all();
-        return view('servidor.create', compact('cargos', 'tipo_usuarios'));
+        $roles = Role::all();
+        return view('servidor.create', compact('cargos', 'roles'));
     }
 
     public function store(StoreServidorRequest $request)
     {
-        $user = User::create(['name' => $request->name,
+        $user = User::create([
+            'name' => $request->name,
             'email' => $request->email,
-            'password' => Hash::make($request->password),
-            'tipo_usuario_id' => $request->tipo_usuario_id]);
-        Servidor::create(['user_id' => $user->id,
+            'password' => Hash::make($request->password)
+
+        ]);
+        
+        $user->roles()->attach($request->role_id);
+
+        Servidor::create(
+            ['user_id' => $user->id,
             'cpf' => $request->cpf,
             'matricula' => $request->matricula,
-            'cargo_id' => $request->cargo_id]);
+            'cargo_id' => $request->cargo_id
+        ]);
+
         return redirect(route('servidor.index'))->with('success', 'Servidor Cadastrado com Sucesso!');
     }
 
     public function edit($servidor_id)
     {
-        $servidor = Servidor::withTrashed()
-            ->find($servidor_id);
+        $servidor = Servidor::withTrashed()->find($servidor_id);
         $cargos = Cargo::all();
-        $tipo_usuarios = TipoUsuario::all();
-        return view('servidor.edit', compact('servidor', 'cargos', 'tipo_usuarios'));
+        $roles = Role::all();
+        return view('servidor.edit', compact('servidor', 'cargos', 'roles'));
     }
 
     public function update(UpdateServidorRequest $request)
     {
         $servidor = Servidor::withTrashed()->find($request->servidor_id);
         $user = $servidor->user;
+
         if ($request->password != null) {
-            $user->update(['name' => $request->name,
+            $user->update([
+                'name' => $request->name,
                 'email' => $request->email,
                 'password' => Hash::make($request->password),
-                'tipo_usuario_id' => $request->tipo_usuario_id]);
+                'role_id' => $request->role_id
```

Comment:
Remover linha `'role_id' => $request->role_id`, as roles não são mais definidas como relação one to many e sim many to many.